domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -190,14 +180,41 @@ private static SecurityDomain toSecurityDomain(final FutureTask<SecurityDomain>
         }
     }
 
+    private static FutureTask<SecurityDomain> toFutureTask(InjectedValue<SecurityDomain> injectedValue) {
+        return new FutureTask<>((Callable<SecurityDomain>)  (() -> injectedValue.getValue()));
+    }
+
     static class IdentityService implements Service<Void> {
 
         private List<FutureTask<SecurityDomain>> getSecurityDomainFutures = Collections.emptyList();
+        private final ManagementSecurityIdentitySupplier securityIdentitySupplier;
+
+        public IdentityService(ManagementSecurityIdentitySupplier securityIdentitySupplier) {
+            this.securityIdentitySupplier = securityIdentitySupplier;
+        }
 
         @Override
         public void start(StartContext context) throws StartException {
             if (getSecurityDomainFutures != null) {
-                getSecurityDomainFutures.forEach(FutureTask::run);
+                Runnable task = () -> {
+                    try {
+                        getSecurityDomainFutures.forEach(FutureTask::run);
+                        List<Supplier<SecurityDomain>> domains = getSecurityDomainFutures.stream().map(f -> (Supplier<SecurityDomain>) (() -> toSecurityDomain(f))).collect(Collectors.toList());
+                        securityIdentitySupplier.setConfiguredSecurityDomainSupplier(domains.get(0));
```

Comment:
IdentityService should have a separate field / setter for this one. It's distinct from the others and doesn't need to be stuck in a list with the others.